packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -223,138 +256,199 @@ class SyncWatcher extends BaseWatcher {
   }
 
   async syncHandler (): Promise<any> {
-    const promises: Array<Promise<any>> = []
+    // Events that are related to user transfers can be polled every cycle while
+    // all other, less time-sensitive events can be polled every N cycles.
+    // If SyncCyclesPerFullSync is 60, then the default values will be 30 seconds for the transferSent
+    // events and 30 minutes for the rest of the events.
+    let promisesPerPoll: EventPromise = []
+    if (
+      !this.isInitialSyncCompleted() ||
+      this.syncIndex % SyncCyclesPerFullSync === 0
```

Comment:
suggestion: for readability maybe this check can be a separate method.

For me it wasn't clear how SyncCyclesPerFullSync worked until i read through the code

Reply:
I think thats a good point. Let me know if you think 5edee2ffe9495b6e85e358fa04d495cd93c85b8f would resolve some of the confusion for future readers.

Reply:
that looks good 👍

a step further would be something like this

```ts
if (this.shouldSyncAllEvents()) {
    promisesPerPoll = this.getAllPromises()
} else {
    promisesPerPoll = this.getTransferSentPromises()
}

// ...

shouldSyncAllEvents(): boolean {
    return !this.isInitialSyncCompleted() || this.isAtNewSyncCyclePoint()
}
  
isAtNewSyncCyclePoint(): boolean {
    return this.syncIndex % SyncCyclesPerFullSync === 0
}
```

Reply:
I like this. My mental model for this kind of stuff has been evolving lately and has been trending more towards this kind of architecture for readability. Good to get your perspective. Resolved with b6f11e9f9c4f906baa78361f9fd97cfba9820293